location/src/main/java/com/yas/location/repository/CountryRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.yas.location.repository;
+
+import com.yas.location.model.Country;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.data.jpa.repository.Query;
+
+@Repository
+public interface CountryRepository extends JpaRepository<Country, Long> {
+
+    @Query("select case when count(c)> 0 then true else false end from Country c where c.name = ?1 and (?2 is null or c.id != ?2)")
+    boolean existsCountryName(String name, Long id);
```

Comment:
confusing method, it's try to check the exists by name, so what's the point of having id.
If having id then we should use jpa built in instead of using Query anotation
```Java
existsByNameAndId(final String name, final Long id)
```

Reply:
For the updating case we don't need to check for the country being updated so we can't use existsByNameAndId